warc_metadata_sidecar.py

```diff
@@ -0,0 +1,261 @@
+            # The payload is how we find the important info. Skip record if empty.
+            if not len(payload.read()):
+                continue
+            if 'text/dns' in record.rec_headers.get_header('Content-Type'):
+                continue
+            record_count += 1
+            record_date = record.rec_headers.get_header('WARC-Date')
+            warcinfo_id = record.rec_headers.get_header('WARC-Warcinfo-ID')
+            warcrecord_id = record.rec_headers.get_header('WARC-Record-ID')
+            # Define specific warc_headers to include in sidecar.
+            warc_dict = {'WARC-Date': record_date, 'WARC-Concurrent-ID': warcrecord_id}
+            if warcinfo_id:
+                warc_dict['WARC-Warcinfo-ID'] = warcinfo_id
+
+            print(url)
+            payload.seek(0)
+            find_mime_and_puid(fido, payload, url)
+
+            if fido.mime:
+                text_format_mimes = re.compile(r'(text|html|xml)')
+                if text_format_mimes.search(fido.mime):
+                    payload.seek(0)
+                    bytes_payload = payload.read()
+                    result_dict = find_character_set(bytes_payload)
+                    lang_cld = find_language(bytes_payload)
+                    text_mime += 1
+                    if not lang_cld:
+                        string_payload = '{0} {1}\n{2} {3}\n{4} {5}'.format(
+                            mime_title, fido.mime,
+                            puid_title, fido.puid,
+                            charset_title, result_dict
+                            )
+                    else:
+                        string_payload = '{0} {1}\n{2} {3}\n{4} {5}\n{6} {7}'.format(
+                            mime_title, fido.mime,
+                            puid_title, fido.puid,
+                            charset_title, result_dict,
+                            language_title, lang_cld
+                            )
+                else:
+                    non_text += 1
+                    string_payload = '{0} {1}\n{2} {3}'.format(
+                        mime_title, fido.mime,
+                        puid_title, fido.puid
+                        )
+            else:
+                non_mime += 1
+                string_payload = ''
+
+            writer = WARCWriter(output, gzip=False)  # TODO: gzip will equal True
+            meta_record = writer.create_warc_record(url,
+                                                    'metadata',
+                                                    payload=io.BytesIO(string_payload.encode()),
+                                                    warc_headers_dict=warc_dict
+                                                    )
+            writer.write_record(meta_record)
+        # Delete sidecar file if the only record is 'warcinfo'.
+        if record_count == 1:
+            if rec_type:
+                os.remove(warc_file_path)
+                logging.info('Deleted sidecar, no records to collect.')
```

Comment:
If we don't add to record_count for warcinfo record, maybe the conditional here would be on `if not record_count`?